spicy/src/compiler/codegen/parser-builder.cc

```diff
@@ -495,6 +498,15 @@ struct ProductionVisitor
         const auto& field = meta.field();
         assert(field); // Must only be called if we have a field.
 
+        if ( ! is_field_owner ) {
```

Comment:
This one is closer to the top so not as confusing, but maybe we should still add a comment here to explain why handling of fields with `! is_field_owner` is not needed and an early return is safe.